### PR TITLE
Mongoferemoveactivity

### PR DESCRIPTION
**Pull request by balajivemo**
I have added all the necessary artifacts for MongoDB Remove Document activity.  Can you please review and approve?
Note that I had changed the activity_test.go for queryDocument (already approved and merged to master) in this check in as well. This is to fix a few issues with the test cases.

### CODE REVIEW

File: mongodb/activity/removeDocument/metadata.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+//ToMap Output
+func (o *Output) ToMap() map[string]interface{} {
+	return map[string]interface{}{
+		"deletedCount": o.DeletedCount,
+	}
+}
+
+//FromMap Output
+func (o *Output) FromMap(values map[string]interface{}) error {
+	o.DeletedCount, _ = (values["deletedCount"].(int64))
```
Comment (skothari-tibco) [inline mongodb/activity/removeDocument/metadata.go]: You can use the `coerce.ToInt64, that will take care of many errors if present and it also has support for number of data types.

Comment (balajivemo) [reply mongodb/activity/removeDocument/metadata.go]: Made this change, tested and pushed the code.

**Review by skothari-tibco — APPROVED** [review]

**Comment by balajivemo** [comment]
@fm-tibco Can you please review and approve this PR ?

File: mongodb/activity/removeDocument/go.mod [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module github.com/project-flogo/datastore-contrib/mongodb/activity/removeDocument
```
Comment (fm-tibco) [inline mongodb/activity/removeDocument/go.mod]: remove go.mod

File: mongodb/activity/removeDocument/go.sum [hunk 3]
```diff
@@ -0,0 +1,57 @@
+github.com/davecgh/go-spew v1.1.0 h1:ZDRjVQ15GmhC3fiQ8ni8+OwkZQO4DARzQgrnXU1Liz8=
```
Comment (fm-tibco) [inline mongodb/activity/removeDocument/go.sum]: remove go.sum

**Review by fm-tibco — CHANGES_REQUESTED** [review]

File: mongodb/activity/removeDocument/go.sum (hunk 3 above)
Comment (balajivemo) [reply mongodb/activity/removeDocument/go.sum]: Done

File: mongodb/activity/removeDocument/go.mod (hunk 2 above)
Comment (balajivemo) [reply mongodb/activity/removeDocument/go.mod]: Done

**Review by fm-tibco — APPROVED** [review]

**Pull request merged**